module/VuFind/src/VuFind/ILS/Logic/AvailabilityStatus.php

```diff
@@ -0,0 +1,287 @@
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  ILS_Logic
+ * @author   Ere Maijala <ere.maijala@helsinki.fi>
+ * @author   Thomas Wagener <wagener@hebis.uni-frankfurt.de>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development Wiki
+ */
+
+namespace VuFind\ILS\Logic;
+
+/**
+ * Availability Status Logic Class
+ *
+ * @category VuFind
+ * @package  ILS_Logic
+ * @author   Ere Maijala <ere.maijala@helsinki.fi>
+ * @author   Thomas Wagener <wagener@hebis.uni-frankfurt.de>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development Wiki
+ */
+class AvailabilityStatus implements AvailabilityStatusInterface
+{
+    /**
+     * Status code for unavailable items
+     *
+     * @var int
+     */
+    public const STATUS_UNAVAILABLE = 0;
+
+    /**
+     * Status code for available items
+     *
+     * @var int
+     */
+    public const STATUS_AVAILABLE = 1;
+
+    /**
+     * Status code for items with uncertain availability
+     *
+     * @var int
+     */
+    public const STATUS_UNCERTAIN = 2;
+
+    /**
+     * Items availability
+     *
+     * @var int
+     */
+    protected int $availability;
+
+    /**
+     * If unknown message shall be used
+     *
+     * @var bool
+     */
+    protected bool $useUnknownMessage = false;
+
+    /**
+     * Constructor
+     *
+     * @param int|bool $availability Availability
+     * @param string   $status       Status Description
+     */
+    public function __construct(int|bool $availability, protected string $status = '')
+    {
+        $this->availability = (int)$availability;
+    }
+
+    /**
+     * Check if available.
+     *
+     * @return bool
+     */
+    public function isAvailable(): bool
+    {
+        return (bool)$this->availability;
+    }
+
+    /**
+     * Check if item has given availability status.
+     *
+     * @param int $availability Availability status
+     *
+     * @return bool
+     */
+    public function is(int $availability): bool
+    {
+        return $this->availability === $availability;
+    }
+
+    /**
+     * Check if status should be visible.
+     *
+     * @return bool
+     */
+    public function isVisible(): bool
+    {
+        return true;
```

Comment:
I assume this is intended as an extension point for customization. Maybe it's worth adding a comment to that effect.